src/JsonSchema/Constraints/CollectionConstraint.php

```diff
@@ -19,10 +19,24 @@
  */
 class CollectionConstraint extends Constraint
 {
-    /**
-     * {@inheritDoc}
-     */
-    public function check($value, $schema = null, JsonPointer $path = null, $i = null)
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function check($value, $schema = null, JsonPointer $path = null, $i = null)
+	{
+		$this->_check($value, $schema, $path, $i);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function coerce(&$value, $schema = null, JsonPointer $path = null, $i = null)
+	{
+		$this->_check($value, $schema, $path, $i, true);
+	}
+
+    protected function _check(&$value, $schema = null, JsonPointer $path = null, $i = null, $coerce = false)
```

Comment:
Interesting idea, but unfortunately any kind of coercion that depends on `check` is a non-starter because I would have to change the first argument to a reference, which *would* break bc.